crates/recursor/src/recursor_pool.rs

```diff
@@ -92,6 +92,7 @@ where
                 let mut options = DnsRequestOptions::default();
                 options.use_edns = security_aware;
                 options.edns_set_dnssec_ok = security_aware;
+                options.recursion_desired = false;
```

Comment:
Can you add some color in a comment on why this is necessary/useful, perhaps rooted in RFC references if possible?